src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1505,50 +1080,26 @@ public bool TryMarkSymbolEndAnalysisComplete(ISymbol symbol, DiagnosticAnalyzer
             SemanticModel model,
             Action<Diagnostic> addDiagnostic,
             Func<Diagnostic, bool> isSupportedDiagnostic,
-            bool isGeneratedCode,
-            OperationAnalyzerStateData? analyzerState)
+            bool isGeneratedCode)
         {
             Debug.Assert(operationActionsByKind != null);
             Debug.Assert(operationActionsByKind.Any());
             Debug.Assert(!isGeneratedCode || !_shouldSkipAnalysisOnGeneratedCode(analyzer));
             Debug.Assert(!IsAnalyzerSuppressedForTree(analyzer, model.SyntaxTree));
 
-            var partiallyProcessedOperation = analyzerState?.CurrentOperation;
-            if (partiallyProcessedOperation != null)
-            {
-                ExecuteOperationActions(partiallyProcessedOperation, operationActionsByKind, containingSymbol, model, addDiagnostic, isSupportedDiagnostic, isGeneratedCode, analyzerState);
-            }
-
-            foreach (var child in operationsToAnalyze)
-            {
-                if (ShouldExecuteOperation(analyzerState, child, analyzer))
-                {
-                    SetCurrentOperation(analyzerState, child);
-
-                    ExecuteOperationActions(child, operationActionsByKind, containingSymbol, model, addDiagnostic, isSupportedDiagnostic, isGeneratedCode, analyzerState);
-                }
-            }
-        }
-
-        private void ExecuteOperationActions(
-            IOperation operation,
-            ImmutableSegmentedDictionary<OperationKind, ImmutableArray<OperationAnalyzerAction>> operationActionsByKind,
-            ISymbol containingSymbol,
-            SemanticModel model,
-            Action<Diagnostic> addDiagnostic,
-            Func<Diagnostic, bool> isSupportedDiagnostic,
-            bool isGeneratedCode,
-            OperationAnalyzerStateData? analyzerState)
-        {
-            if (operationActionsByKind.TryGetValue(operation.Kind, out var actionsForKind))
+            foreach (var operation in operationsToAnalyze)
             {
-                foreach (var action in actionsForKind)
+                if (ShouldExecuteOperation(operation, analyzer))
                 {
-                    ExecuteOperationAction(action, operation, containingSymbol, model, addDiagnostic, isSupportedDiagnostic, isGeneratedCode, analyzerState);
+                    if (operationActionsByKind.TryGetValue(operation.Kind, out var actionsForKind))
```

Comment:
❔ It seems like it's more likely for `actionsForKind` to be empty than for `ShouldExecuteOperation` to be false. Should we reorder these checks and avoid calling `ShouldExecuteOperation` on operations that don't have any registered actions?

Reply:
➡️ Submitted as follow-up #66923